tools/main.cpp

```diff
@@ -70,7 +70,18 @@ int main(int argc, char** argv) {
           antler::update_project,
           antler::validate_project> runner{app};
 
-   CLI11_PARSE(app, argc, argv);
+   try {
+      app.parse(argc, argv);
+   }
+   // This hack fix parsing error handling of an empty command argument. Example: antler-proj build ~/path/proj ""
+   catch (CLI::ExtrasError &e) {
+      if (std::string_view{"The following argument was not expected: "} == e.what()) {
+         return app.exit(CLI::ExtrasError("Empty argument is encountered in the command line", e.get_exit_code()));
+      }
```

Comment:
> According to https://github.com/AntelopeIO/antler-proj/blob/main/tools/CLI11.hpp#L902 ExtrasError always contain either `The following arguments were` or `The following arguments was`

But we should not change the behavior in case if this exception thrown for non empty arguments. In this case all these arguments should be printed as "The following arguments were not expected: arg1 arg2 etc." or "The following argument was not expected: arg"
We should change the message only in case if it prints "The following argument was not expected: " which happens only if we have one "" (empty argument) in the command line.

Reply:
Ah, I see. This is bigger hack than I thought.
Isn't there any other way to fix it?
I.e. checking arguments before parsing if any of them is `""`

Reply:
These are "extra" arguments - so they aren't caught by the standard validators.

I've suggested a change (below) that could address this.

Reply:
> Ah, I see. This is bigger hack than I thought. Isn't there any other way to fix it? I.e. checking arguments before parsing if any of them is `""`

This is the best solution which I can invent without changing of the CLI11.hpp code.
The problem is not with empty arguments of options, the problem with empty options. 
The parser recognize the subcommand, then find all its options and their arguments, and all other text which it can find the parser consider as "extra" arguments, i.e. trash which it shows in the text of the exception when prints the error. All it works if these extra arguments have symbols. But it seems authors of the parser don't know that it possible to declare an empty argument in the command line, so the parser just don't handle this case at all.